### PR TITLE
Bids 2417/wrong user input errors

### PR DESCRIPTION
**Pull request by Eisei24**
### <samp>🤖 Generated by Copilot at e4dc883</samp>

This pull request improves the error handling, logging, validation, and performance of various handlers and queries in the eth2-beaconchain-explorer project. It also fixes some typos, unused code, and route parameters. The main files affected are `handlers/dashboard.go`, `handlers/epoch.go`, `handlers/eth1Deposits.go`, `handlers/graffitiwall.go`, `handlers/pools_rocketpool.go`, `handlers/slot.go`, `handlers/user.go`, `handlers/validator.go`, `handlers/validatorRewards.go`, `handlers/validators_slashings.go`, `handlers/validators.go`, `cmd/explorer/main.go`, `db/db.go`, `handlers/common.go`, `handlers/epochs.go`, `handlers/eth1Blocks.go`, `handlers/eth1tx.go`, `handlers/eth2Deposits.go`, `handlers/slots.go`, and `handlers/validators_leaderboard.go`.

### CODE REVIEW

**Review by LuccaBitfly — CHANGES_REQUESTED** [review]

File: handlers/validator.go [hunk 1]
```diff
@@ -243,7 +244,7 @@ func Validator(w http.ResponseWriter, r *http.Request) {
 	} else {
 		// Request came with a validator index number
 		index, err = strconv.ParseUint(vars["index"], 10, 64)
-		if err != nil {
+		if err != nil || index > math.MaxInt32 { // index in postgres is limited to int
```
[the remark this answers is not in the document]
Comment (Eisei24) [reply handlers/validator.go:247]: But max uint32 is higher than max int32.

[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply handlers/validator.go:247]: IIRC there are no unsigned integers (or we don't use them as types for columns), so its max value is max int32. Correct me if I'm wrong.

File: cmd/explorer/main.go [hunk 2]
```diff
@@ -467,7 +467,7 @@ func main() {
 			router.HandleFunc("/validator/{pubkey}/deposits", handlers.ValidatorDeposits).Methods("GET")
 			router.HandleFunc("/validator/{index}/slashings", handlers.ValidatorSlashings).Methods("GET")
 			router.HandleFunc("/validator/{index}/effectiveness", handlers.ValidatorAttestationInclusionEffectiveness).Methods("GET")
-			router.HandleFunc("/validator/{pubkey}/save", handlers.ValidatorSave).Methods("POST")
+			router.HandleFunc("/validator/save", handlers.ValidatorSave).Methods("POST")
```
[the remark this answers is not in the document]
Comment (Eisei24) [reply cmd/explorer/main.go:470]: While this should be idempotent PUT can't be used in HTML forms (as discussed) so I left it at POST.
The rest was changed as desired.

File: handlers/validator.go [hunk 3]
```diff
@@ -1383,7 +1384,7 @@ func ValidatorSave(w http.ResponseWriter, r *http.Request) {
```
[the remark this answers is not in the document]
Comment (Eisei24) [reply handlers/validator.go:1384]: Changed to `ValidatorName`.

File: handlers/validator.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Eisei24) [reply handlers/validator.go:247]: Yes they don't exist, that is why we need to check that it is within int32 here.
If you search for eg. 4,000,000,000 in postgres for an int column an error will be thrown.

[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply handlers/validator.go:247]: Sorry my bad, i meant `strconv.ParseInt`

File: handlers/validator.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply handlers/validator.go:1386]: Maybe a little more descriptive? Something like `SaveValidatorName` or `SetValidatorName`.

File: cmd/explorer/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply cmd/explorer/main.go]: Seems fine, have you tested changing a name with this?

[the remark this answers is not in the document]
Comment (Eisei24) [reply cmd/explorer/main.go]: Yes, it worked.

File: handlers/validator.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Eisei24) [reply handlers/validator.go:1386]: Sure, renamed to `SaveValidatorName`.

**Review by LuccaBitfly — APPROVED** [review]
lgtm

**Pull request merged**